modules/lang-painless/src/main/java/org/elasticsearch/painless/CompilerSettings.java

```diff
@@ -19,10 +19,18 @@
 
 package org.elasticsearch.painless;
 
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Setting.Property;
+
 /**
  * Settings to use when compiling a script.
  */
 public final class CompilerSettings {
+    /**
+     * Are regexes enabled? This is a node level settings because regexes break out of painless's lovely sandbox and can cause stack
```

Comment:
Yes, I'll change to `setting`.

I'd certainly love for regexes to be less brittle about this, yeah. Lucene's regexes are interesting because they have protection but they don't offer all the features we'd need. I'd love to take a crack at exposing them because they'd be safe but I'd like to get this in first.